airflow/models/xcom.py

```diff
@@ -55,6 +56,17 @@ class BaseXCom(Base, LoggingMixin):
     task_id = Column(String(ID_LEN, **COLLATION_ARGS), primary_key=True)
     dag_id = Column(String(ID_LEN, **COLLATION_ARGS), primary_key=True)
 
+    # For _now_, we link this via execution_date, in 2.3 we will migrate this table to use run_id too
+    dag_run = relationship(
+        "DagRun",
+        primaryjoin="""and_(
+            BaseXCom.dag_id == foreign(DagRun.dag_id),
+            BaseXCom.execution_date == foreign(DagRun.execution_date)
+        )""",
```

Comment:
Is this passed as string for late evaluation?

Reply:
Yes, to avoid circular imports